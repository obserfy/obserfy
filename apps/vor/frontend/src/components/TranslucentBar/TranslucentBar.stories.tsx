import { Meta, Story } from "@storybook/react"
import { Box, Flex } from "theme-ui"
import TranslucentBar, { TranslucentBarProps } from "./TranslucentBar"

export default {
  title: "Core/TranslucentBar",
  component: TranslucentBar,
  parameters: {
    componentSubtitle: "Just a simple TranslucentBar",
  },
} as Meta

const Template: Story<TranslucentBarProps> = () => {
  return (
    <Box>
      <TranslucentBar
        boxSx={{
          position: "fixed",
          top: 0,
          left: 0,
          right: 0,
        }}
      >
        <Flex sx={{ justifyContent: "space-around" }}>
          <Box m={3}>One</Box>
          <Box m={3}>Two</Box>
          <Box m={3}>Three</Box>
        </Flex>
      </TranslucentBar>
      <Flex>
        <Box backgroundColor="red" sx={{ height: 300, width: 100 }} />
        <Box backgroundColor="yellow" sx={{ height: 300, width: 100 }} />
        <Box backgroundColor="green" sx={{ height: 300, width: 100 }} />
        <Box backgroundColor="blue" sx={{ height: 300, width: 100 }} />
      </Flex>
      <Flex>
        <Box backgroundColor="blue" sx={{ height: 300, width: 100 }} />
        <Box backgroundColor="red" sx={{ height: 300, width: 100 }} />
        <Box backgroundColor="green" sx={{ height: 300, width: 100 }} />
        <Box backgroundColor="yellow" sx={{ height: 300, width: 100 }} />
      </Flex>
      <Flex>
        <Box backgroundColor="red" sx={{ height: 300, width: 100 }} />
        <Box backgroundColor="yellow" sx={{ height: 300, width: 100 }} />
        <Box backgroundColor="blue" sx={{ height: 300, width: 100 }} />
        <Box backgroundColor="green" sx={{ height: 300, width: 100 }} />
      </Flex>
    </Box>
  )
}

export const Default = Template.bind({})
Default.args = {}
