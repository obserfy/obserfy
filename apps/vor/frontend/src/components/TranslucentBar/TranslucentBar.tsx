import { FC } from "react"
import { Box, ThemeUIStyleObject } from "theme-ui"

export interface TranslucentBarProps {
  as?: React.ElementType
  boxSx?: ThemeUIStyleObject
}

export const TranslucentBar: FC<TranslucentBarProps> = ({
  boxSx = {},
  as,
  children,
}) => {
  // const { deviceMemory } = useMemoryStatus({
  //   deviceMemory: 0, // default always disables blur
  // })
  //
  const sx = { zIndex: 100, ...boxSx }
  // if (deviceMemory > 3) {
  // Only enable blur on mid to hi end devices, cause it is expensive.
  // sx["@supports (backdrop-filter: blur(20px))"] = {
  //   backgroundColor: "surfaceBlurTransparent",
  //   backdropFilter: "saturate(180%) blur(20px)",
  // }
  // }

  return (
    <Box as={as} backgroundColor="surfaceBlurNonTransparent" sx={sx}>
      {children}
    </Box>
  )
}

export default TranslucentBar
