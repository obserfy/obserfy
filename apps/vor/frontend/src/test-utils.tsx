/* eslint-disable import/no-extraneous-dependencies */
import { FC, ReactElement } from "react"
import { QueryClient, QueryClientProvider } from "react-query"
import { ThemeProvider, Box, Theme } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import { I18nProvider } from "@lingui/react"
import { i18n } from "@lingui/core"
import { en } from "make-plural/plurals"
import theme from "./gatsby-plugin-theme-ui"
// @ts-ignore
import enCatalog from "../i18n/lingui/en/messages.js"

// any used here because the catalogs are generated b lingui
// not our responsibility
i18n.loadLocaleData("en", { plurals: en })
i18n.load("en", enCatalog.messages as any)
i18n.activate("en")

const queryClient = new QueryClient()
const history = createHistory(createMemorySource("/"))

const AllProviders: FC = ({ children }) => {
  return (
    <I18nProvider i18n={i18n}>
      <QueryClientProvider client={queryClient}>
        <LocationProvider history={history}>
          <ThemeProvider theme={theme as Theme}>
            <Box sx={{ fontSize: [16, 20] }}>{children}</Box>
          </ThemeProvider>
        </LocationProvider>
      </QueryClientProvider>
    </I18nProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
