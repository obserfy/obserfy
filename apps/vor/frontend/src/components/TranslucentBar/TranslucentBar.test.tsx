import { render } from "../../test-utils"
import { Default } from "./TranslucentBar.stories"

describe("TranslucentBar", () => {
  it("should render correctly", () => {
    const { container } = render(<Default />)
    expect(container).toMatchSnapshot()
  })
})
