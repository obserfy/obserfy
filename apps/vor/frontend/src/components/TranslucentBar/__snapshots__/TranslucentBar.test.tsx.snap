// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TranslucentBar should render correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: var(--theme-ui-colors-surfaceBlurNonTransparent);
  z-index: 100;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -ms-flex-pack: space-around;
  -webkit-justify-content: space-around;
  justify-content: space-around;
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin: 16px;
}

.emotion-7 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-8 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: red;
  height: 300px;
  width: 100px;
}

.emotion-9 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: yellow;
  height: 300px;
  width: 100px;
}

.emotion-10 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: green;
  height: 300px;
  width: 100px;
}

.emotion-11 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: blue;
  height: 300px;
  width: 100px;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <div
          class="emotion-3"
        >
          <div
            class="emotion-4"
          >
            One
          </div>
          <div
            class="emotion-4"
          >
            Two
          </div>
          <div
            class="emotion-4"
          >
            Three
          </div>
        </div>
      </div>
      <div
        class="emotion-7"
      >
        <div
          class="emotion-8"
        />
        <div
          class="emotion-9"
        />
        <div
          class="emotion-10"
        />
        <div
          class="emotion-11"
        />
      </div>
      <div
        class="emotion-7"
      >
        <div
          class="emotion-11"
        />
        <div
          class="emotion-8"
        />
        <div
          class="emotion-10"
        />
        <div
          class="emotion-9"
        />
      </div>
      <div
        class="emotion-7"
      >
        <div
          class="emotion-8"
        />
        <div
          class="emotion-9"
        />
        <div
          class="emotion-11"
        />
        <div
          class="emotion-10"
        />
      </div>
    </div>
  </div>
</div>
`;
